packages/material-ui/src/internal/SwitchBase.js

```diff
@@ -30,128 +29,117 @@ export const styles = {
 /**
  * @ignore - internal component.
  */
-class SwitchBase extends React.Component {
-  constructor(props) {
-    super();
-    this.isControlled = props.checked != null;
-    this.state = {};
-    if (!this.isControlled) {
-      // not controlled, use internal state
-      this.state.checked = props.defaultChecked !== undefined ? props.defaultChecked : false;
-    }
-  }
-
-  handleFocus = event => {
-    if (this.props.onFocus) {
-      this.props.onFocus(event);
+const SwitchBase = React.forwardRef(function SwitchBase(props, ref) {
+  const {
+    autoFocus,
+    checked: checkedProp,
+    checkedIcon,
+    classes,
+    className: classNameProp,
+    defaultChecked,
+    disabled: disabledProp,
+    icon,
+    id,
+    inputProps,
+    inputRef,
+    muiFormControl,
```

Comment:
Good opportunity to introduce a `useFormControlContext` and get rid of the hoc. Up to you if you want to do this in this PR or as a follow up.

Reply:
I've left all the other hooks, but I'll keep this in mind :)